documentation/adoc/cluster-controller.adoc

```diff
@@ -543,22 +516,95 @@ data:
   image: "strimzi/kafka-connect:latest"
   healthcheck-delay: "60"
   healthcheck-timeout: "5"
-  KAFKA_CONNECT_BOOTSTRAP_SERVERS: "my-cluster-kafka:9092"
-  KAFKA_CONNECT_GROUP_ID: "my-connect-cluster"
-  KAFKA_CONNECT_KEY_CONVERTER: "org.apache.kafka.connect.json.JsonConverter"
-  KAFKA_CONNECT_KEY_CONVERTER_SCHEMAS_ENABLE: "true"
-  KAFKA_CONNECT_VALUE_CONVERTER: "org.apache.kafka.connect.json.JsonConverter"
-  KAFKA_CONNECT_VALUE_CONVERTER_SCHEMAS_ENABLE: "true"
-  KAFKA_CONNECT_CONFIG_STORAGE_REPLICATION_FACTOR: "3"
-  KAFKA_CONNECT_OFFSET_STORAGE_REPLICATION_FACTOR: "3"
-  KAFKA_CONNECT_STATUS_STORAGE_REPLICATION_FACTOR: "3"
+  connect-config: |-
+    {
+      "bootstrap.servers": "my-cluster-kafka:9092",
+      "group.id": "my-connect-cluster",
+      "offset.storage.topic": "my-connect-cluster-offsets",
+      "config.storage.topic": "my-connect-cluster-configs",
+      "status.storage.topic": "my-connect-cluster-status",
+      "key.converter": "org.apache.kafka.connect.json.JsonConverter",
+      "value.converter": "org.apache.kafka.connect.json.JsonConverter",
+      "key.converter.schemas.enable": true,
+      "value.converter.schemas.enable": true,
+      "internal.key.converter": "org.apache.kafka.connect.json.JsonConverter",
+      "internal.value.converter": "org.apache.kafka.connect.json.JsonConverter",
+      "internal.key.converter.schemas.enable": false,
+      "internal.value.converter.schemas.enable": false,
+      "config.storage.replication.factor": 3,
+      "offset.storage.replication.factor": 3,
+      "status.storage.replication.factor": 3
+    }
 ----
 
 The resources created by the cluster controller into the Kubernetes/OpenShift cluster will be the following :
 
 * [connect-cluster-name]-connect Deployment which is in charge to create the Kafka Connect worker node pods
 * [connect-cluster-name]-connect Service which exposes the REST interface for managing the Kafka Connect cluster
 
+[[kafka_connect_configuration_json_config]]
+===== Kafka Connect configuration
+
+`connect-config` field allows detailed configuration of Apache Kafka Connect and Connect S2I. This field should contain
+a JSON object with Kafka Connect configuration options as keys. The values could be in one of the following types:
+
+* String
+* Integer
+* Long
```

Comment:
These are JSON types, which only has `Number`, not separate types for different precisions nor a distinction between floating point and integer. Maybe go so far as to sat "one of the following JSON types". Or you could say "any of the JSON scalar types" and I think the meaning would be clear.